### PR TITLE
Issue #111 Implement UI to add and edit templates

### PR DESCRIPTION
**Pull request by dmitriim**

### CODE REVIEW

File: amd/src/managetemplates.js [hunk 1]
```diff
@@ -0,0 +1,68 @@
+/**
+ * Tour management code.
```
Comment (nhoobin) [inline amd/src/managetemplates.js]: Update doc header

File: amd/src/managetemplates.js [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+ * @class managetemplates
+ * @package quizaccess_seb
+ * @copyright  2020 Dmitrii Metelkin <dmitriim@catalyst-au.net>
+ */
+define(
+    ['jquery', 'core/ajax', 'core/str', 'core/notification'],
+    function($, ajax, str, notification) {
+        var manager = {
+            /**
+             * Confirm removal of the specified tour.
```
Comment (nhoobin) [inline amd/src/managetemplates.js]: Update doc header

File: amd/src/managetemplates.js [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                            window.location = targetUrl;
+                        });
+
+                        return;
+                    })
+                    .catch();
+            },
+
+            /**
+             * Setup the tour management UI.
```
Comment (nhoobin) [inline amd/src/managetemplates.js]: Update doc header

File: amd/src/managetemplates.js [hunk 4]
```diff
@@ -0,0 +1,68 @@
...
+             * @method setup
+             */
+            setup: function() {
+                $('body').delegate('[data-action="delete"]', 'click', manager.removeTemplate);
+            }
+        };
+
+        return /** @alias module:quizaccess_seb/managetemplates */ {
+            /**
+             * Setup the tour management UI.
```
Comment (nhoobin) [inline amd/src/managetemplates.js]: Update doc header

File: lang/en/quizaccess_seb.php [hunk 5]
```diff
@@ -166,3 +166,19 @@
...
+$string['edittemplate'] = 'Edit template';
+$string['content'] = 'Template';
+$string['name'] = 'Name';
+$string['description'] = 'Description';
+$string['enabled'] = 'Enabled';
+$string['name'] = 'Name';
+$string['duplicatetemplate'] = "Template with the same name already exists";
+$string['invalidtemplate'] = "Invalid SEB config template";
+$string['cantdelete'] = 'Template can\'t be deleted as it has been used for one or more quizzes';
+$string['confirmtemplateremovaltitle'] = 'Conform template removal?';
```
Comment (nhoobin) [inline lang/en/quizaccess_seb.php]:
```suggestion
$string['confirmtemplateremovaltitle'] = 'Confirm template removal?';
```

File: amd/src/managetemplates.js (hunk 1 above)
Comment (dmitriim) [reply amd/src/managetemplates.js]: fixed

File: lang/en/quizaccess_seb.php (hunk 5 above)
Comment (dmitriim) [reply lang/en/quizaccess_seb.php]: fixed

File: classes/form/template.php [hunk 6]
```diff
@@ -0,0 +1,114 @@
...
+        // Check name.
+        if (empty($data->name)) {
+            $newerrors['name'] = get_string('namerequired', 'quizaccess_seb');
+        }
+
+        // Check submitted template as a text.
+        if (!empty($this->get_persistent()->get('id'))) {
+            $plist = new CFPropertyList();
+            try {
+                $plist->parse($data->content);
```
Comment (nhoobin) [inline classes/form/template.php]: With full Moodle debugging enabled, 

CFProperyList Parse throws a warning if $data->content is empty
`Warning: DOMDocument::loadXML(): Empty string supplied as input`

If the field is not valid XML it also warns
`Warning: DOMDocument::loadXML(): Start tag expected, '<' not found in Entity`

We have some tests which over this regarding the config key generation, test_config_key_not_generated_with_bad_xml

File: classes/helper.php [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+
+    /**
+     * Get a filler icon for display in the actions column of a table.
+     *
+     * @param string $url The URL for the icon.
+     * @param string $icon The icon identifier.
+     * @param string $alt The alt text for the icon.
+     * @param string $iconcomponent The icon component.
+     * @param array $options Display options.
+     * @return  string
```
Comment (nhoobin) [inline classes/helper.php]:
```suggestion
     * @return string
```

File: classes/form/template.php (hunk 6 above)
Comment (dmitriim) [reply classes/form/template.php]: I have added custom error handler.

File: classes/helper.php (hunk 7 above)
Comment (dmitriim) [reply classes/helper.php]: fixed

**Pull request merged**